contracts/child/ChildChainManager/ChildChainManager.sol

```diff
@@ -76,15 +81,18 @@ contract ChildChainManager is IChildChainManager, Initializable, AccessControl {
     }
 
     function _syncDeposit(bytes memory syncData) private {
-        (address user, address rootToken, bytes memory depositData) = abi
-            .decode(syncData, (address, address, bytes));
+        (address user, address rootToken, bytes memory depositData, address callback) = abi
+            .decode(syncData, (address, address, bytes, address));
         address childTokenAddress = rootToChildToken[rootToken];
         require(
             childTokenAddress != address(0x0),
             "ChildChainManager: TOKEN_NOT_MAPPED"
         );
         IChildToken childTokenContract = IChildToken(childTokenAddress);
         childTokenContract.deposit(user, depositData);
+        if (callback != address(0)) {
+            IDepositCallback(callback).processSyncDeposit(user, rootToken, depositData);
+        }
```

Comment:
We also need to make sure we limit this call with proper gas to avoid it since it will be a system call. This method will be called from here: https://github.com/maticnetwork/genesis-contracts/blob/master/contracts/StateReceiver.sol#L24-L34

For the initial call (in-state receiver), we have limit of 5m. So this should be < (5m - other execution -- Line 84-92).